src/runtime/opencl/opencl_common.h

```diff
@@ -229,6 +229,8 @@ class OpenCLWorkspace : public DeviceAPI {
   cl_context context{nullptr};
   // whether the workspace it initialized.
   bool initialized_{false};
+  // whether the workspace is in profiling mode.
+  bool profiling{false};
```

Comment:
Just another idea. It is not critical. You can get the properties on the queue and check if the profiling was enabled or not. In this case, it is not necessary to introduce this variable.
Note: you can request properties by using [clGetCommandQueueInfo](https://www.khronos.org/registry/OpenCL/sdk/1.1/docs/man/xhtml/clGetCommandQueueInfo.html)